packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/s3.ts

```diff
@@ -0,0 +1,39 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as firehose from '@aws-cdk/aws-kinesisfirehose';
+import { CfnDeliveryStream } from '@aws-cdk/aws-kinesisfirehose';
+import * as s3 from '@aws-cdk/aws-s3';
+import { Construct } from 'constructs';
+
+/**
+ * Props for defining an S3 destination of a Kinesis Data Firehose delivery stream.
+ */
+export interface S3Props extends firehose.DestinationProps { }
+
+/**
+ * An S3 bucket destination for data from a Kinesis Data Firehose delivery stream.
+ */
+export class S3 extends firehose.DestinationBase {
+  constructor(private readonly bucket: s3.IBucket, s3Props: S3Props = {}) {
+    super(s3Props);
+  }
+
+  bind(scope: Construct, options: firehose.DestinationBindOptions): firehose.DestinationConfig {
+    return {
+      properties: {
+        extendedS3DestinationConfiguration: this.createExtendedS3DestinationConfiguration(scope, options.deliveryStream),
+      },
+    };
+  }
+
+  private createExtendedS3DestinationConfiguration(
+    scope: Construct,
+    deliveryStream: firehose.IDeliveryStream,
+  ): CfnDeliveryStream.ExtendedS3DestinationConfigurationProperty {
+    this.bucket.grantReadWrite(deliveryStream);
+    return {
+      cloudWatchLoggingOptions: this.createLoggingOptions(scope, deliveryStream, 'S3Destination'),
+      roleArn: (deliveryStream.grantPrincipal as iam.IRole).roleArn,
```

Comment:
Updated the PR to pass in an `IRole` into the `DestinationBindOptions`. Still open for discussion on why we should create the role specifically in the integration class.  

If I create a new role for the integration, then there will be at least two roles associated with this delivery stream, because it will still need a role at the DeliveryStream level to be able to decrypt keys for server-side encryption for example. And then I would need to manage multiple roles for one DeliveryStream and it would get somewhat complicated.

Reply:
I think that's fine to have 2 Roles. If you allow passing a Role also to the integration classes (probably by adding it to the `DestinationProps` interface), a customer can decide to use a single Role for both.